src/components/AddBooks/AddBook.js

```diff
@@ -0,0 +1,174 @@
+import React, { useState } from "react";
+import axios from "axios";
+import Navigation from "../Navigation/Navigation";
+import Grid from "../Grid/Grid";
+
+const AddBooks = () => {
+  const [title, setTitle] = useState("");
+  const [author, setAuthor] = useState("");
+  const [quantity, setQuantity] = useState("");
+  const [search, setSearch] = useState("");
+  const [books, setBooks] = useState([]);
+  const [selectedBook, setSelectedBook] = useState(null);
+  const [orderQuantity, setOrderQuantity] = useState(1);
+  const [orderSuccess, setOrderSuccess] = useState(false);
+  const [orderError, setOrderError] = useState(false);
+
+  const handleTitleChange = (event) => {
+    setTitle(event.target.value);
+  };
+
+  const handleAuthorChange = (event) => {
+    setAuthor(event.target.value);
+  };
+
+  const handleQuantityChange = (event) => {
+    setQuantity(event.target.value);
+  };
+
+  const handleSearch = () => {
+    axios
+      .get(`https://www.googleapis.com/books/v1/volumes?q=${search}`)
+      .then((res) => {
+        setBooks(res?.data?.items);
+      })
+      .catch((err) => console.log(err));
+  };
+
+  const handleBookClick = (book) => {
+    setSelectedBook(book);
+    setOrderQuantity(1);
+  };
+
+  const handleOrderQuantityChange = (event) => {
+    setOrderQuantity(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    // Add logic here to handle the form submission, such as sending data to a backend server or updating state variables.
+    if (selectedBook) {
+      const newBook = {
```

Comment:
this needs all of the criteria found in firebase so every key needs to be represented here we cant pass have filled in books